benchmarks/benchmarks/spatial.py

```diff
@@ -54,6 +54,61 @@ def time_build(self, mnr, cls_name):
             self.cls(self.data)
 
 
+class PresortedDataSetup(Benchmark):
+    params = [
+        [(3, 10 ** 4, 1000), (8, 10 ** 4, 1000), (16, 10 ** 4, 1000)],
+        [True, False],
+        ['random', 'sorted'],
+        [0.5]
+    ]
+    param_names = ['(m, n, r)', 'balanced', 'order', 'radius']
+
+    def setup(self, mnr, balanced, order, radius):
+        m, n, r = mnr
+
+        np.random.seed(1234)
+        self.data = {
+            'random': np.random.uniform(size=(n, m)),
+            'sorted': np.repeat(np.arange(n, 0, -1)[:, np.newaxis],
+                                m,
+                                axis=1) / n
+        }
+
+        self.queries = np.random.uniform(size=(r, m))
+        self.T = cKDTree(self.data.get(order), balanced_tree=balanced)
+
+
+class BuildUnbalanced(PresortedDataSetup):
+    params = PresortedDataSetup.params[:-1]
+    param_names = PresortedDataSetup.param_names[:-1]
+
+    def setup(self, *args):
+        super().setup(*args, None)
+
+    def time_build(self, mnr, balanced, order):
+        cKDTree(self.data.get(order), balanced_tree=balanced)
+
+
+class QueryUnbalanced(PresortedDataSetup):
+
+    def setup(self, *args):
+        super().setup(*args, None)
```

Comment:
Something is going wrong here:
```
· Discovering benchmarks
· Running 4 total benchmarks (1 commits * 1 environments * 4 benchmarks)
[  0.00%] ·· Benchmarking existing-py_home_rgommers_anaconda3_bin_python
[ 12.50%] ··· Running (spatial.BuildUnbalanced.time_build--)....
[ 62.50%] ··· spatial.BuildUnbalanced.time_build                                      ok
[ 62.50%] ··· =================== ========== ============= =============
              --                                        order           
              ------------------------------ ---------------------------
                   (m, n, r)       balanced      random        sorted   
              =================== ========== ============= =============
                (3, 10000, 1000)     True     2.28±0.03ms   1.29±0.01ms 
                (3, 10000, 1000)    False     1.31±0.02ms     874±10μs  
                (8, 10000, 1000)     True     2.79±0.04ms   1.75±0.02ms 
                (8, 10000, 1000)    False     1.77±0.05ms   1.27±0.03ms 
               (16, 10000, 1000)     True     3.34±0.07ms   2.23±0.01ms 
               (16, 10000, 1000)    False     2.22±0.01ms   1.79±0.02ms 
              =================== ========== ============= =============

[ 75.00%] ··· spatial.QueryUnbalanced.time_query                                  failed
[ 75.00%] ··· =================== ========== ============== ==============
              --                                     order / radius       
              ------------------------------ -----------------------------
                   (m, n, r)       balanced   random / 0.5   sorted / 0.5 
              =================== ========== ============== ==============
                (3, 10000, 1000)     True        failed         failed    
                (3, 10000, 1000)    False        failed         failed    
                (8, 10000, 1000)     True        failed         failed    
                (8, 10000, 1000)    False        failed         failed    
               (16, 10000, 1000)     True        failed         failed    
               (16, 10000, 1000)    False        failed         failed    
              =================== ========== ============== ==============

[ 75.00%] ···· For parameters: (3, 10000, 1000), True, 'random', 0.5
               Traceback (most recent call last):
                 File "/home/rgommers/anaconda3/lib/python3.7/site-packages/asv/benchmark.py", line 1184, in main_run_server
                   main_run(run_args)
                 File "/home/rgommers/anaconda3/lib/python3.7/site-packages/asv/benchmark.py", line 1052, in main_run
                   skip = benchmark.do_setup()
                 File "/home/rgommers/anaconda3/lib/python3.7/site-packages/asv/benchmark.py", line 583, in do_setup
                   result = Benchmark.do_setup(self)
                 File "/home/rgommers/anaconda3/lib/python3.7/site-packages/asv/benchmark.py", line 515, in do_setup
                   setup(*self._current_params)
                 File "/home/rgommers/code/scipy/benchmarks/benchmarks/spatial.py", line 100, in setup
                   super().setup(*args, None)
               TypeError: setup() takes 5 positional arguments but 6 were given
...               
```

This setup looks the same as the one for `BuildUnbalanced`, but `params`, `param_names` treatment is different.